docs/https_source_interface.rst

```diff
@@ -4,45 +4,93 @@ HTTPS on the Source Interface
 The *Source Interface* for SecureDrop is automatically served over
 as a Tor Hidden Service, requiring a ``*.onion`` URL to access it.
 While Tor Hidden Services provide end-to-end encryption by default, as well
-as strong anonymity, high-profile organizations may wish to provide
-further verification to Sources that the SecureDrop instance belongs
-to the organization listed on the *Landing Page*.
+as strong anonymity, there are several reasons why you might want to consider
+deploying an additional layer of encryption and authentication via HTTPS:
+
+* The cryptographic primitives used by Tor Hidden Services are considered to be
+  outdated, and while there are no known compromises of the security of Tor
+  Hidden Services due to this issue, it may be wise to provide an additional
+  layer of transport encryption using stronger cryptographic primitives, which
+  is most easily achieved by setting up HTTPS on the Source Interface.
+
+  .. note:: This issue is being addressed by the Tor Project with their Next
+     Generation Onion Services design, but that design is still a work in
+     progress and is not expected to be deployed until December 2017 at the
+     soonest.
+
+* Extended Validation (EV) certificates, which are currently the only type of
+  certificates that may be issued for ``*.onion`` addresses, are intended to
+  attest to the identity of the organization running a service. This provides
+  an additional measure of authenticity (in addition to the organization's
+  *Landing Page* and the `SecureDrop Directory`_) to help assure sources that
+  they are communicating with the intended organization when they access a
+  given Source Interface.
```

Comment:
> While an additional layer of encryption is a nice bonus, if we really believed that there may be a break in onion service crypto in the wild and that this was a likely vector of attack, then we'd make HTTPS on all interfaces a requirement and not optional.

All I can say here is that the fact that Tor Hidden Service's crypto primitives would be considered insecure if the system were being designed today has been and continues to be a strong motivation for the administrators who have already deployed HTTPS on their SecureDrops. Even if there is not a break right now, this is still a concern in the context of adversaries who may be storing Tor traffic with the aim of trying to break the crypto at a later date.

I think it would great to make HTTPS on all SecureDrop's required, or at least strongly recommended. The only reasons why we haven't done so already are:

1. It wasn't even possible to get a cert for an .onion until 2015
2. After the initial .onion certs were issued, there was a lot of debate as to whether they would be approved by the relevant standards bodies, and this approval process took a long time. We decided to wait and see because we didn't want to tell administrators to spend money on certificates, only to find that they would stop being supported.
3. Maintaining an HTTPS deployment over time requires a significant amount of technical know-how and attention from the SecureDrop instance administrators. They need to pay attention to cert expiry and a whole host of other issues. Due to SecureDrop's decentralized architecture, we were leery of encouraging people to do this unless we felt that were capable of managing the lifecycle of their HTTPS deployment on their own.

Finally, the Tor Project is aware of this issue, have already improved the cryptographic primitives in some areas of the Tor protocol due to this exact concern, and hidden services will start using a modern set of primitives once the next generation design is deployed. HTTPS is pretty much just a stop-gap until the next-generation design is available, but I still think it is worthwhile.

So: I think this potential motivation for deploying HTTPS on a SecureDrop instance should be kept in because it is a meaningful defense-in-depth measure.

> In my view the reason why we want to allow HTTPS on the source interface is the second reason you describe here: to demonstrate to sources that they are communicating with the onion service that is indeed controlled by the news organization they want to talk to.

In theory, yes, but of course EV certificates have a whole host of problems: most people don't know what they mean, the EV validation process has had issues in the past, and there are generally trust issues in the context of Certificate Authorities. So I think this is a helpful additional way to authenticate a SecureDrop instance that complements the other methods we've already developed (landing page, directory, printing it in the print publication), but I do not think of it as the primary motivation to deploy HTTPS on the Source Interface.